cmd/buildah/images.go

```diff
@@ -299,35 +304,64 @@ func outputImages(ctx context.Context, systemContext *types.SystemContext, store
 			imageID = shortID(image.ID)
 		}
 
+		repos := imageReposToMap(image.Names)
+		if len(repos) == 0 {
+			repos = map[string][]string{"": []string{""}}
+		}
 	outer:
-		for name, tags := range imageReposToMap(image.Names) {
-			for _, tag := range tags {
-				if !matchesReference(name+":"+tag, argName) {
+		for repo, tagsAndDigests := range repos {
+			var imageDigests []string
+			for _, tagOrDigest := range tagsAndDigests {
+				imageName := ""
+				imageDigest := ""
+				imageTag := ""
+				if repo != "" && tagOrDigest != "" {
+					if digest.Digest(tagOrDigest).Validate() == nil {
+						imageDigest = tagOrDigest
+						imageName = repo + "@" + imageDigest
+					} else {
+						imageTag = tagOrDigest
+						if len(image.Digest) > 0 {
+							imageDigest = image.Digest.String()
+						}
+						imageName = repo + ":" + imageTag
+					}
+				}
+
+				if !matchesReference(imageName, argName) {
```

Comment:
BTW does `matchesReference` now need updating? If I’m reading it right, `imageName: "foo@sha256:…"` doesn’t match `argName: "foo"`.

Reply:
Yeah, it does need an update.

Reply:
Updated it.